src/dll.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""File implements a simple doubly linked list in python."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """Class implements Nodes for use in our dll."""
+    def __init__(self, data=None, nxt=None, prev=None):
+        self.data = data
+        self._next = nxt
+        self._prev = prev
+
+
+class DoublyLinkedList(object):
+    """Class implements a simple doubly linked list."""
+    def __init__(self, iterable=None):
+        self.head = None
+        self.tail = None
+        try:
+            if iterable:
+                for item in iterable:
+                    self.push(item)
+        except TypeError:
+            raise TypeError("Please enter an object that is iterable.")
+
+    def push(self, data):
+        """Insert data at the head of the list."""
+        new_node = Node(data)
```

Comment:
Since you guys have (smartly) implemented your `Node.__init__()` with keyword arguments for its next and previous attributes, you can and should just put those values in as arguments to __init__() instead of setting them after initialization.